src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1055,16 +1055,71 @@ public void Connect(IPAddress[] addresses, int port)
                 throw new NotSupportedException(SR.net_invalidversion);
             }
 
+// Disable CS0162: Unreachable code detected
+//
+// SuportsMultipleConnectAttempts is a constant; when false, the following lines will trigger CS0162.
+#pragma warning disable 162
             Exception lastex = null;
-            foreach (IPAddress address in addresses)
+            if (SocketPal.SupportsMultipleConnectAttempts)
```

Comment:
This if/else seems like something that could be pushed down into the PAL, e.g. ConnectCore or TryConnectAllEndpoints or something like that.